### PR TITLE
Release 2.3.5

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: ktor-client/ktor-client-android/api/ktor-client-android.api [hunk 1]
```diff
@@ -8,7 +8,6 @@ public final class io/ktor/client/engine/android/AndroidClientEngine : io/ktor/c
 	public fun execute (Lio/ktor/client/request/HttpRequestData;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
 	public synthetic fun getConfig ()Lio/ktor/client/engine/HttpClientEngineConfig;
 	public fun getConfig ()Lio/ktor/client/engine/android/AndroidEngineConfig;
-	public fun getDispatcher ()Lkotlinx/coroutines/CoroutineDispatcher;
```
Comment (Stexxe) [inline ktor-client/ktor-client-android/api/ktor-client-android.api:11]: Isn't this an API change?

File: ktor-client/ktor-client-core/common/src/io/ktor/client/engine/HttpClientEngineConfig.kt [hunk 2]
```diff
@@ -14,6 +14,10 @@ public open class HttpClientEngineConfig {
     /**
      * Specifies network threads count advice.
      */
+    @Deprecated(
+        "The [threadsCount] property is deprecated. The [Dispatchers.IO] is used by default.",
```
Comment (Stexxe) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/engine/HttpClientEngineConfig.kt:18]: Is it okay to have a deprecation error in a patch release?

File: ktor-client/ktor-client-plugins/ktor-client-logging/api/ktor-client-logging.api [hunk 3]
```diff
@@ -5,6 +5,7 @@ public final class io/ktor/client/plugins/logging/LogLevel : java/lang/Enum {
 	public static final field INFO Lio/ktor/client/plugins/logging/LogLevel;
 	public static final field NONE Lio/ktor/client/plugins/logging/LogLevel;
 	public final fun getBody ()Z
+	public static fun getEntries ()Lkotlin/enums/EnumEntries;
```
Comment (Stexxe) [inline ktor-client/ktor-client-plugins/ktor-client-logging/api/ktor-client-logging.api:8]: New API in a patch release

File: ktor-http/api/ktor-http.api [hunk 4]
```diff
@@ -40,6 +40,7 @@ public final class io/ktor/http/CacheControl$NoStore : io/ktor/http/CacheControl
 public final class io/ktor/http/CacheControl$Visibility : java/lang/Enum {
 	public static final field Private Lio/ktor/http/CacheControl$Visibility;
 	public static final field Public Lio/ktor/http/CacheControl$Visibility;
+	public static fun getEntries ()Lkotlin/enums/EnumEntries;
```
Comment (Stexxe) [inline ktor-http/api/ktor-http.api:43]: new API

File: ktor-io/posix/src/io/ktor/utils/io/ByteReadChannelNative.kt [hunk 5]
```diff
@@ -75,12 +75,14 @@ public actual interface ByteReadChannel {
...
      */
+    @OptIn(ExperimentalForeignApi::class)
     public suspend fun readAvailable(dst: CPointer<ByteVar>, offset: Int, length: Int): Int
 
     /**
      * Reads all available bytes to [dst] buffer and returns immediately or suspends if no bytes available
      * @return number of bytes were read or `-1` if the channel has been closed
      */
+    @OptIn(ExperimentalForeignApi::class)
     public suspend fun readAvailable(dst: CPointer<ByteVar>, offset: Long, length: Long): Int
```
Comment (Stexxe) [inline ktor-io/posix/src/io/ktor/utils/io/ByteReadChannelNative.kt:86]: is it okay to have an OptIn annotation in a patch release?

**Review by Stexxe — CHANGES_REQUESTED** [review]

File: ktor-client/ktor-client-core/common/src/io/ktor/client/engine/HttpClientEngineConfig.kt (hunk 2 above)
Comment (e5l) [reply ktor-client/ktor-client-core/common/src/io/ktor/client/engine/HttpClientEngineConfig.kt:18]: Let me make it as warning

File: ktor-client/ktor-client-plugins/ktor-client-logging/api/ktor-client-logging.api (hunk 3 above)
Comment (e5l) [reply ktor-client/ktor-client-plugins/ktor-client-logging/api/ktor-client-logging.api:8]: getEntries is default enum field generated by the kotlin compiler

File: ktor-http/api/ktor-http.api (hunk 4 above)
Comment (e5l) [reply ktor-http/api/ktor-http.api:43]: same

File: ktor-io/posix/src/io/ktor/utils/io/ByteReadChannelNative.kt (hunk 5 above)
Comment (e5l) [reply ktor-io/posix/src/io/ktor/utils/io/ByteReadChannelNative.kt:86]: yep, it's kotlin native and triggered by the kotlin version

File: ktor-client/ktor-client-android/api/ktor-client-android.api (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-android/api/ktor-client-android.api:11]: Nope, we dropped the override field but still have it in the base

**Review by marychatte — APPROVED** [review]
lgtm

**Review by Stexxe — APPROVED** [review]

**Pull request merged**